### PR TITLE
[@types/twitch-js] Add type definitions

### PR DESCRIPTION
**Pull request by Roundaround**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/twitch-js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    init: (channel: string) => void;
+    enter: (channel: string, username: string) => void;
+    leave: (channel: string, username: string) => boolean;
+    pick: (channel: string) => string|null;
+    reset: (channel: string) => void;
+    count: (channel: string) => number;
+    isParticipating: (channel: string, username: string) => boolean;
+}
+
+export class Client extends EventEmitter {
```
Comment (sheetalkamat) [inline types/twitch-js/index.d.ts]: this should be client and not Client as per https://github.com/twitch-apis/twitch-js/blob/master/docs/Examples.md#browser

Comment (sheetalkamat) [inline types/twitch-js/index.d.ts]: this should be client and not Client as per https://github.com/twitch-apis/twitch-js/blob/master/docs/Examples.md#browser

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/twitch-js/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    getUsername(): string;
+    handleMessage(message: Message): void;
+    isMod(channel: string, username: string): boolean;
+    readyState(): 'CONNECTING'|'OPEN'|'CLOSING'|'CLOSED';
+
+    /// Commands
+    action(channel: string, message: string): Promise<[string, string]>;
+    ban(channel: string, username: string, reason: string): Promise<[string, string, string]>;
+    clear(channel: string): Promise<[string]>;
+    color(channel: string, newColor: string): Promise<[string]>;
```
Comment (sheetalkamat) [inline types/twitch-js/index.d.ts]: This seems incorrect too https://github.com/twitch-apis/twitch-js/blob/master/docs/Chat/Commands.md#color

File: types/twitch-js/index.d.ts (hunk 1 above)
Comment (Roundaround) [reply types/twitch-js/index.d.ts]: It actually exports both, so I was going with the one that fit TS naming standards. Can I do something like `export type client = Client;` (or vice-versa)?

File: types/twitch-js/index.d.ts (hunk 2 above)
Comment (Roundaround) [reply types/twitch-js/index.d.ts]: In this case the documentation is actually wrong. I'll submit a PR there for it. From the source: `function color(channel, newColor)`

**Comment by Roundaround** [comment]
Closing in favor of local type definitions.

**Pull request closed without merging**